docs/reference/api/lightning_adapter.txt

```diff
@@ -0,0 +1,138 @@
+<https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html>`_,
+please read more about :ref:`PyTorchTrial <pytorch-trial>` and
+:ref:`experiment-configuration`.
+
+Porting your ``PyTorchLightning`` code is often pretty simple: 1. Bring
+in your ``LightningModule`` and ``LightningDataModule`` and initialize
+them 2. Create a new trial based on ``LightningAdapter`` and initialize
+it. 3. Define the dataloaders.
+
+Here is an example:
+
+.. code:: python
+
+   from determined.pytorch import PyTorchTrialContext, DataLoader
+   from determined.pytorch.lightning import LightningAdapter
+
+   # bring in your LightningModule and optinally LightningDataModule
+   from mnist import LightningMNISTClassifier, MNISTDataModule
+
+
+   class MNISTTrial(LightningAdapter):
+       def __init__(self, context: PyTorchTrialContext) -> None:
+           # instantiate your LightningModule with hyperparameter from the Determined
+           # config file or from the searcher for automatic hyperparameter tuning.
+           lm = LightningMNISTClassifier(lr=context.get_hparam("learning_rate"))
+
+           # instantiate your LightningDataModule and make it distributed training ready.
+           data_dir = f"/tmp/data-rank{context.distributed.get_rank()}"
+           self.dm = MNISTDataModule(context.get_data_config()["url"], data_dir)
+
+           # initialize LightningAdapter.
+           super().__init__(context, lightning_module=lm)
+           self.dm.prepare_data()
+
+       def build_training_data_loader(self) -> DataLoader:
+           self.dm.setup()
+           dl = self.dm.train_dataloader()
+           return DataLoader(
+               dl.dataset, batch_size=dl.batch_size, num_workers=dl.num_workers
+           )
+
+       def build_validation_data_loader(self) -> DataLoader:
+           self.dm.setup()
+           dl = self.dm.val_dataloader()
+           return DataLoader(
+               dl.dataset, batch_size=dl.batch_size, num_workers=dl.num_workers
+           )
+
+.. autoclass:: determined.pytorch.lightning.LightningAdapter
+   :members:
+   :exclude-members: trial_controller_class, trial_context_class, evaluate_full_dataset, train_batch, evaluate_batch, setup_optimizers_schedulers
+   :inherited-members:
+   :member-order: bysource
+   :special-members: __init__
+
+In this approach, the ``LightningModule`` is not paired with the
+``Trainer`` so that there are some methods and hooks that are not
+supported. Read about those here:
+
+-  No separate test-set definition in Determined: ``test_step``,
+   ``test_step_end``, ``test_epoch_end``, ``on_test_batch_start``,
+   ``on_test_batch_end``, ``on_test_epoch_start``,
+   ``on_test_epoch_end``, ``test_dataloader``.
+
+-  No fit or pre-train stage: ``setup``, ``teardown``, ``on_fit_start``,
+   ``on_fit_end``, ``on_pretrain_routine_start``,
+   ``on_pretrain_routine_end``.
+
+-  And: ``training_step_end`` & ``validation_step_end``, ``hiddens``
```

Comment:
Just to make it clear that these are also not supported, I might replace "And:" -> "Additionally, no"